### PR TITLE
Make IOBuf clearer work with run_until_complete

### PR DESCRIPTION
**Pull request by andrewgait**
Another iobuf issue, this time noticed by @lplana in PDP2.  This fixes the issue with the ChipIOBufClearer when using run_until_complete(...) - I think it also works properly with run forever (i.e. run(None)) but it would be good if there are more tests for this other than a single GFE example...

Tested by https://github.com/SpiNNakerManchester/SpiNNakerGraphFrontEnd/pull/143 and https://github.com/SpiNNakerManchester/sPyNNaker8/pull/402

### CODE REVIEW

**Comment by alan-stokes** [comment]
fair enough. can see why that would bugger it up. but why have both a class variable and it being handed down to _run? keep one and lose the other i would have thought

**Comment by andrewgait** [comment]
> fair enough. can see why that would bugger it up. but why have both a class variable and it being handed down to _run? keep one and lose the other i would have thought

Will do this once @lplana has confirmed this fixes his issue...

**Comment by andrewgait** [comment]
Fixed to use class variable rather than passing value down through functions.

Started testing on SpiNNakerManchester/SpiNNakerGraphFrontEnd#143 and SpiNNakerManchester/sPyNNaker8#402.

File: spinn_front_end_common/interface/abstract_spinnaker_base.py [hunk 1]
```diff
@@ -279,6 +279,9 @@ class AbstractSpinnakerBase(ConfigHandler, SimulatorInterface):
         # The lowest values auto pause resume may use as steps
         "_minimum_auto_time_steps",
 
+        #
```
Comment (alan-stokes) [inline spinn_front_end_common/interface/abstract_spinnaker_base.py:282]: docs

File: spinn_front_end_common/interface/abstract_spinnaker_base.py [hunk 2]
```diff
@@ -492,6 +495,7 @@ def __init__(
         self._no_machine_time_steps = None
         self._minimum_auto_time_steps = self._config.getint(
                 "Buffers", "minimum_auto_time_steps")
+        self._run_until_complete = False
```
Comment (alan-stokes) [inline spinn_front_end_common/interface/abstract_spinnaker_base.py:498]: id personally move that above the things where we start reading configs. but your welcome to ignore this one

File: spinn_front_end_common/interface/abstract_spinnaker_base.py [hunk 3]
```diff
@@ -1931,7 +1934,7 @@ def _do_run(self, n_machine_time_steps, graph_changed, run_until_complete):
             reraise(*e_inf)
 
     def _create_execute_workflow(
-            self, n_machine_time_steps, graph_changed, run_until_complete):
+            self, n_machine_time_steps, graph_changed):
```
Comment (alan-stokes) [inline spinn_front_end_common/interface/abstract_spinnaker_base.py:1937]: realign with def. as fits on 1 line now

**Comment by alan-stokes** [comment]
after that. i consider this done

**Comment by alan-stokes** [comment]
sorry bud. conflict.

**Comment by andrewgait** [comment]
> after that. i consider this done

All done now I think... ?

**Pull request merged**